shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/repositories/OnboardingRepository.kt

```diff
@@ -0,0 +1,60 @@
+package com.mbta.tid.mbta_app.repositories
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringSetPreferencesKey
+import co.touchlab.skie.configuration.annotations.DefaultArgumentInterop
+import com.mbta.tid.mbta_app.model.OnboardingScreen
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+interface IOnboardingRepository {
+    suspend fun getPendingOnboarding(): List<OnboardingScreen>
+
+    suspend fun markOnboardingCompleted(screen: OnboardingScreen)
+}
+
+class OnboardingRepository : IOnboardingRepository, KoinComponent {
```

Comment:
thought (non-blocking): we haven't done this consistently, but I think it would be worth landing on a naming convention for repositories that use other repositories to help with maintaining clear layers / hierarchy. We could adopt UseCase more consistently for that.